modules/ppcp-wc-gateway/src/Endpoint/ReturnUrlEndpoint.php

```diff
@@ -130,7 +161,95 @@ function( $allowed_hosts ) : array {
 			wp_safe_redirect( $success['redirect'] );
 			exit();
 		}
+
+		wc_add_notice( __( 'Payment processing failed. Please try again or contact support.', 'woocommerce-paypal-payments' ), 'error' );
 		wp_safe_redirect( wc_get_checkout_url() );
 		exit();
 	}
+
+	/**
+	 * Check if order needs 3DS completion.
+	 *
+	 * @param mixed $order The PayPal order.
+	 * @return bool
+	 */
+	private function needs_3ds_completion( $order ): bool {
```

Comment:
```suggestion
	 * @param Order $order The PayPal order.
	 * @return bool
	 */
	private function needs_3ds_completion( Order $order ): bool {
```

I think this should have a [type](https://github.com/woocommerce/woocommerce-paypal-payments/blob/9ef34dc7737e575afac27dcd8673f2fdd6080440/modules/ppcp-api-client/src/Entity/Order.php#L15), otherwise how do we use `$order->status()` ?